crates/fuel-core/src/state.rs

```diff
@@ -12,18 +9,18 @@ use std::{
 };
 
 pub type DataSource = Arc<dyn TransactableStorage>;
-pub type ColumnId = u32;
-pub type KVItem = DatabaseResult<(Vec<u8>, Vec<u8>)>;
+pub type Value = Arc<Vec<u8>>;
+pub type KVItem = DatabaseResult<(Vec<u8>, Value)>;
 
 pub trait KeyValueStore {
-    fn get(&self, key: &[u8], column: Column) -> DatabaseResult<Option<Vec<u8>>>;
+    fn get(&self, key: &[u8], column: Column) -> DatabaseResult<Option<Value>>;
```

Comment:
One downside to forcing this to be inside an arc is that the caller cannot modify the `Vec<u8>` if they need to. Rocks does not return an `Arc` and if we move to a rocks transaction then it also won't have an `Arc` so it's probably best to not force the `Arc` on the caller.

Reply:
The caller is deserializer, so we don't modify it and only use it in read only mode